libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+public class PolicyManager {
+
+    protected final Policy defaultPolicy;
+    protected final Map<String, Policy> pluginPolicies;
+
+    public PolicyManager(Policy defaultPolicy, Map<String, Policy> pluginPolicies) {
+        this.defaultPolicy = Objects.requireNonNull(defaultPolicy);
+        this.pluginPolicies = Collections.unmodifiableMap(Objects.requireNonNull(pluginPolicies));
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Just curious, why do we need equals/hashcode on the manager?

Reply:
I removed them for now